cadquery/hull.py

```diff
@@ -195,33 +200,38 @@ def _pt_arc(p: Point, a: Arc) -> Tuple[float, float, float, float]:
 
     return x1, y1, x2, y2
 
-
-def pt_arc(p: Point, a: Arc) -> Tuple[float, Segment]:
+def pt_arc(angle: float, p: Point, a: Arc) -> Tuple[float, Segment]:
 
     x, y = p.x, p.y
     x1, y1, x2, y2 = _pt_arc(p, a)
 
+    if ((x1, y1) == (x2, y2)):
+        return inf, Segment(Point(inf, inf), Point(inf, inf))
+
     angles = atan2p(x1 - x, y1 - y), atan2p(x2 - x, y2 - y)
     points = Point(x1, y1), Point(x2, y2)
-    ix = int(argmin(angles))
+    ix = int(argmin([a + 2*pi if a < angle else a for a in angles]))
 
     return angles[ix], Segment(p, points[ix])
 
 
-def arc_pt(a: Arc, p: Point) -> Tuple[float, Segment]:
+def arc_pt(angle: float, a: Arc, p: Point) -> Tuple[float, Segment]:
 
     x, y = p.x, p.y
     x1, y1, x2, y2 = _pt_arc(p, a)
 
+    if ((x1, y1) == (x2, y2)):
+        return inf, Segment(Point(inf, inf), Point(inf, inf))
+
     angles = atan2p(x - x1, y - y1), atan2p(x - x2, y - y2)
     points = Point(x1, y1), Point(x2, y2)
 
-    ix = int(argmax(angles))
+    ix = int(argmax([a + 2*pi if a < angle else a for a in angles]))
```

Comment:
When the angle loops around back past 0 radians, this caused the argmin/argmax of `pt_arc` and `arc_pt` to choose the wrong side of the arc to continue from.